### PR TITLE
Upgrade RejectsOrders

### PR DESCRIPTION
**Pull request by reaperrr**
Adds explicit `Rejects` list and makes `RejectsOrders` multi-trait compatible.

Split from #12924.

### CODE REVIEW

**Review by pchote — APPROVED** [review]
Carrying over my :+1: from #12924.

File: OpenRA.Mods.Common/Traits/RejectsOrders.cs [hunk 1]
```diff
@@ -10,21 +10,26 @@
...
 namespace OpenRA.Mods.Common.Traits
 {
 	[Desc("Can be used to make a unit partly uncontrollable by the player.")]
 	public class RejectsOrdersInfo : ConditionalTraitInfo
 	{
-		[Desc("Possible values include Attack, AttackMove, Guard, Move.")]
+		[Desc("Explicit list of rejected orders. Leave empty to reject all minus those listed under Except.")]
+		public readonly HashSet<string> Reject = new HashSet<string>();
+
+		[Desc("List of orders that should *not* be rejected.")]
```
Comment (GraionDilach) [inline OpenRA.Mods.Common/Traits/RejectsOrders.cs]: "Also overrides other instances of this trait's Reject fields."

I feel that fact worthy to be documented. With that, I gladly  :+1: this. (The solution itself is top-notch, but this aspect might end up overlooked at first.)

**Comment by reaperrr** [comment]
Updated.

**Review by GraionDilach — APPROVED** [review]
:+1:

**Comment by atlimit8** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/8d2b35c1a931a75a6d85edf6ebaa436049275f43...b61502b0e1476de5ee105a372dd07de4c624b9d9)

**Pull request merged**